code/game/atoms.dm

```diff
@@ -109,6 +116,86 @@
 		return 1
 	return
 
+// Helper for adding verbs with timers.
+/atom/proc/add_verb(the_verb, datum/callback/callback)
+	if (callback && !callback.Invoke())
+		return
+
+	verbs += the_verb
+
+#define NOT_FLAG(flag) (!(flag & use_flags))
+#define HAS_FLAG(flag) (flag & use_flags)
+
+// Checks if user can use this object. Set use_flags to customize what checks are done.
+// Returns 0 if they can use it, a value representing why they can't if not.
+// Flags are in `code/__defines/misc.dm`
+/atom/proc/use_check(mob/user, use_flags = 0, show_messages = FALSE)
+	. = 0
+	if (NOT_FLAG(USE_ALLOW_NONLIVING) && !isliving(user))
+		// No message for ghosts.
+		return USE_FAIL_NONLIVING
+
+	if (NOT_FLAG(USE_ALLOW_NON_ADJACENT) && !Adjacent(user))
+		if (show_messages)
+			to_chat(user, "<span class='notice'>You're too far away from [src] to do that.</span>")
+		return USE_FAIL_NON_ADJACENT
+
+	if (NOT_FLAG(USE_ALLOW_DEAD) && user.stat == DEAD)
+		if (show_messages)
+			to_chat(user, "<span class='notice'>How do you expect to do that when you're dead?</span>")
+		return USE_FAIL_DEAD
+
+	if (NOT_FLAG(USE_ALLOW_INCAPACITATED) && (user.incapacitated()))
+		if (show_messages)
+			to_chat(user, "<span class='notice'>You cannot do that in your current state.</span>")
+		return USE_FAIL_INCAPACITATED
+
+	if (NOT_FLAG(USE_ALLOW_NON_ADV_TOOL_USR) && !user.IsAdvancedToolUser())
+		if (show_messages)
+			to_chat(user, "<span class='notice'>You don't know how to operate [src].</span>")
+		return USE_FAIL_NON_ADV_TOOL_USR
+
+	if (HAS_FLAG(USE_DISALLOW_SILICONS) && issilicon(user))
+		if (show_messages)
+			to_chat(user, "<span class='notice'>How do you propose doing that without hands?</span>")
+		return USE_FAIL_IS_SILICON
+
+	if (HAS_FLAG(USE_FORCE_SRC_IN_USER) && !(src in user))
+		if (show_messages)
+			to_chat(user, "<span class='notice'>You need to be holding [src] to do that.</span>")
+		return USE_FAIL_NOT_IN_USER
+
+/atom/proc/use_check_and_message(mob/user, use_flags = 0)
+	. = use_check(user, use_flags, TRUE)
+
+#undef NOT_FLAG
+#undef HAS_FLAG
+
+/atom/proc/get_light_and_color(var/atom/origin)
+	if(origin)
+		color = origin.color
+		set_light(origin.light_range, origin.light_power, origin.light_color)
+
+/atom/proc/find_up_hierarchy(var/atom/target)
```

Comment:
Do we need this *and* `/proc/get()` they seem really similar?

Reply:
`find_up_hierarchy` is used in just one place, to check validity of modifier items by checking if it's still in the inventory (or same loc I guess). But it checks a specific `/atom` while `get()` checks just a type, so technically they are different (though we could make one do both if we really wanted).

I'd rather not touch it though, better not break it, lol.